src/OrchardCore.Modules/OrchardCore.AuditTrail/Handlers/GlobalContentHandler.cs

```diff
@@ -0,0 +1,103 @@
+
+
+        public GlobalContentHandler(
+            IYesSqlSession session,
+            IHttpContextAccessor hca,
+            ILogger<GlobalContentHandler> logger,
+            IAuditTrailManager auditTrailManager,
+            IEnumerable<IAuditTrailContentEventHandler> auditTrailEvents)
+        {
+            _hca = hca;
+            _session = session;
+            _auditTrailManager = auditTrailManager;
+            _auditTrailEvents = auditTrailEvents;
+
+            Logger = logger;
+        }
+
+
+        public override Task UpdatedAsync(UpdateContentContext context) =>
+            // The Document.Id of the child content items will always be 0. We don't need to log the children.
+            // In this case we can prevent to log the items inside the FlowPart, BagPart, MenuItemLink, etc.
+            context.ContentItem.Id != 0 ?
+                RecordAuditTrailEventAsync(ContentAuditTrailEventProvider.Saved, context.ContentItem) :
+                Task.CompletedTask;
+
+        public override Task CreatedAsync(CreateContentContext context)
+        {
+            if (_hca.HttpContext.Items.ContainsKey("OrchardCore.AuditTrail.Restored"))
+            {
+                return RecordAuditTrailEventAsync(ContentAuditTrailEventProvider.Restored, context.ContentItem);
+            }
+            else if (!context.ContentItem.Published)
+            {
+                return RecordAuditTrailEventAsync(ContentAuditTrailEventProvider.Created, context.ContentItem);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        public override Task PublishedAsync(PublishContentContext context) =>
+            RecordAuditTrailEventAsync(ContentAuditTrailEventProvider.Published, context.ContentItem);
+
+        public override Task UnpublishedAsync(PublishContentContext context) =>
+            RecordAuditTrailEventAsync(ContentAuditTrailEventProvider.Unpublished, context.ContentItem);
+
+        public override Task RemovedAsync(RemoveContentContext context) =>
+            RecordAuditTrailEventAsync(ContentAuditTrailEventProvider.Removed, context.ContentItem);
+
+        public override Task ClonedAsync(CloneContentContext context) =>
+            RecordAuditTrailEventAsync(ContentAuditTrailEventProvider.Cloned, context.ContentItem);
+
+
+        private async Task RecordAuditTrailEventAsync(string eventName, IContent content)
+        {
+            var buildingAuditTrailEventContext = new BuildingAuditTrailEventContext(content.ContentItem, eventName);
+
+            await _auditTrailEvents.InvokeAsync((provider, context) =>
+                provider.BuildingAuditTrailEventAsync(context), buildingAuditTrailEventContext, Logger);
+
+            // -1 is a unique contentItem.Id that only Preview is using such that another stored document can't have
+            // the same one in the IContentManagerSession index.
+            if (buildingAuditTrailEventContext.IsCanceled || content.ContentItem.Id == -1) return;
```

Comment:
Just for infos I am trying to formalize that an item is in preview mode on this pr here https://github.com/OrchardCMS/OrchardCore/pull/6626

Possibly here it needs to make the check before invoking the events, that the item is being built in preview mode?

Reply:
You are right, making the check before invoking the events is more optimal. I will do that now and after your mentioned PR is merged I will try to use your `ContentPreviewFeature` class in the module.

Reply:
Did you implement this, @domonkosgabor? I don't see it happening.